src/plots/cartesian/dragbox.js

```diff
@@ -125,10 +125,11 @@ function makeDragBox(gd, plotinfo, x, y, w, h, ns, ew) {
         yActive = isDirectionActive(yaxes, ns);
         allFixedRanges = !yActive && !xActive;
 
-        links = calcLinks(gd, gd._fullLayout._axisConstraintGroups, xaHash, yaHash);
         matches = calcLinks(gd, gd._fullLayout._axisMatchGroups, xaHash, yaHash);
-        editX = ew || links.isSubplotConstrained || matches.isSubplotConstrained;
-        editY = ns || links.isSubplotConstrained || matches.isSubplotConstrained;
+        links = calcLinks(gd, gd._fullLayout._axisConstraintGroups, xaHash, yaHash, matches);
+        var spConstrained = links.isSubplotConstrained || matches.isSubplotConstrained;
+        editX = ew || spConstrained;
+        editY = ns || spConstrained;
```

Comment:
This whole section is badly in need of a rewrite, but I didn't do it here, I just patched it up to work - fixing some existing bugs that are particularly evident in `axes_scaleanchor-with-matches`. Ideally we would make it much more explicit what's going on, rather than the very indirect actions happening in `dragAxList`, `updateMatchedAxRange`, `updateSubplots`, and `ticksAndAnnotations`.